tests/test_helpers.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import unittest
+import mock
+import pytest
+
+from greenswitch import esl, helpers
+
+
+@pytest.mark.usefixtures("outbound_session")
+@pytest.mark.usefixtures("linger_disconnect_event")
+class TestWhileConnected(unittest.TestCase):
+    def setUp(self):
+        self.outbound_session.connect()
+        self.execute_slow_task = mock.MagicMock()
+
+    def simulate_caller_hangup(self):
+        self.outbound_session.handle_event(self.linger_disconnect_event)
+
+    def test_context_manager(self):
+        with self.assertRaises(esl.OutboundSessionHasGoneAway):
+            with helpers.while_connected(self.outbound_session):
```

Comment:
easier to use without passing the outbound_session as an argument. I would prefer implementing this inside the outbound session.

Reply:
Done